### PR TITLE
Bugfix/typos in docs

### PR DESCRIPTION
**Pull request by mikemorency**
This fixes a ton of typos and grammatical errors throughout the collection. I used the document generated by QE to guide the changes.

I also tried to update the provision_vm documentation to be a bit more readable since that was noted as being particularly difficult by QE. I think its because there are so many options that were copied from the community module, leading to weird syntax.

I also updated our role dependency docs to match other validated content collections. They use that section to identify pre-req steps, system requirements, and other miscellaneous dependencies rather than python or collection deps.

### CODE REVIEW

**Comment by anna-savina** [comment]
@mikemorency Why didn't you specify dependencies for roles?
```
Dependencies
N/A
```

**Comment by anna-savina** [comment]
cluster_settings role:

https://github.com/redhat-cop/cloud.vmware_ops/pull/104/files#diff-4b3a77ab2c4520eb346a208302798319f1e9b04629125d05638f9aaf2459029dR86

Suggestion:
```suggestion
  - str, Indicates whether VMs should be powered off if a host determines that it is isolated from the rest of the compute resource. One of `none`, `powerOff`, `powerOn`
```

https://github.com/redhat-cop/cloud.vmware_ops/pull/104/files#diff-4b3a77ab2c4520eb346a208302798319f1e9b04629125d05638f9aaf2459029dR30

Suggestion:
```suggestion
  - str or int, The port used to authenticate to the vSphere vCenter that contains the cluster to configure.
```

**Comment by anna-savina** [comment]
content_library role:

https://github.com/redhat-cop/cloud.vmware_ops/pull/104/files#diff-b79baabe2c4381009ee3fdb5a3357da78cdffd0f562090e6c1508664e5debb44R49

```suggestion
- **content_library_ssl_thumbprint**:
```

File: roles/deploy_ovf/README.md [hunk 1]
```diff
@@ -2,31 +2,31 @@
...
 
 - **deploy_ovf_port**:
-  - str or int, The port to use to authenticate to the vSphere vCenter or ESXI host.
+  - str or int, The port to use to authenticate to the vSphere vCenter or ESXi host.
 
 
 ### Placement
 - **deploy_ovf_cluster_name**:
-  - str, The name of the cluster in vSphere vCenter to configure. Required if your connecting to a vcenter cluster and `deploy_ovf_esxi_host` is not provided.
+  - str, The name of the cluster in vSphere vCenter to configure. Required if your connecting to a vCenter cluster and `deploy_ovf_esxi_host` is not provided.
```
Comment (anna-savina) [inline roles/deploy_ovf/README.md:29]:
```suggestion
  - str, The name of the cluster in vSphere vCenter to configure. Required if you're connecting to a vCenter cluster and `deploy_ovf_esxi_host` is not provided.
```

File: roles/deploy_ovf/README.md [hunk 2]
```diff
@@ -2,31 +2,31 @@
...
 - **deploy_ovf_hostname**:
-  - str, Required. The hostname or IP address of the vSphere vCenter or ESXI host.
+  - str, Required. The hostname or IP address of the vSphere vCenter or ESXi host.
 
 - **deploy_ovf_validate_certs**
   - bool, Allows connection when SSL certificates are not valid. Set to false when certificates are not trusted.
 
 - **deploy_ovf_port**:
-  - str or int, The port to use to authenticate to the vSphere vCenter or ESXI host.
+  - str or int, The port to use to authenticate to the vSphere vCenter or ESXi host.
```
Comment (anna-savina) [inline roles/deploy_ovf/README.md:24]:
```suggestion
  - str or int, The port used to authenticate to the vSphere vCenter that contains the cluster to configure.
```

**Comment by anna-savina** [comment]
deploy_ovf:

https://github.com/redhat-cop/cloud.vmware_ops/pull/104/files#diff-e8b805f5c57b32c56b5deab96655d02081262a8294697a55e1830ed7d15ad09bR151

```suggestion
      deploy_ovf_library: my-content-library
```

File: roles/export_vm_as_ovf/README.md [hunk 3]
```diff
@@ -1,8 +1,8 @@
 # export_vm_as_ovf
 
-A role to export a VM from VCenter or ESXi as an OVF. The VM is exported to the local filesystem of the host running the tasks (anisble_host).
+A role to export a VM from vCenter or ESXi as an OVF. The VM is exported to the local filesystem of the host running the tasks (anisble_host).
```
Comment (anna-savina) [inline roles/export_vm_as_ovf/README.md:3]:
```suggestion
A role to export a VM from vCenter or ESXi as an OVF. The VM is exported to the local filesystem of the host running the tasks (ansible_host).
```

**Comment by anna-savina** [comment]
export_vm_as_ovf role:

https://github.com/redhat-cop/cloud.vmware_ops/pull/104/files#diff-2ac5bb4d54196e20a8f6f9f556f51c999b00a38e05b1a044ab77e93b346e0a91R76

```suggestion
- name: Export VM To Localhost
```

File: roles/provision_vcenter/README.md [hunk 4]
```diff
@@ -88,17 +88,14 @@ pyvomi
   - str, The IP family that the VM should use for its network. ipv4 or ipv6. Default is ipv4
 
 - **provision_vcenter_vm_deployment_option**:
-  - str, The deployment "size" for the vcenter appliance. See the VMWare documentation for the accepted values and their requirements. Default is tiny.
+  - str, The deployment "size" for the vCenter appliance. See the VMWare documentation for the accepted values and their requirements. Default is tiny.
```
Comment (anna-savina) [inline roles/provision_vcenter/README.md:91]:
```suggestion
  - str, The deployment "size" for the vCenter appliance. See the VMware documentation for the accepted values and their requirements. Default is tiny.
```

File: roles/snapshot_management/README.md [hunk 5]
```diff
@@ -22,108 +23,108 @@ N/A
...
 
-- **snapshot_management_quiesce**: 
-  - If set to C(true) and virtual machine is powered on, it will quiesce the file system in the virtual machine.
+- **snapshot_management_quiesce**:
+  - If set to `true` and the virtual machine is powered on, it will quiesce the file system in the virtual machine.
   - Note that VMware Tools are required for this flag.
-  - If virtual machine is powered off or VMware Tools are not available, then this flag is set to C(false).
-  - If virtual machine does not provide capability to take quiesce snapshot, then this flag is set to C(false).
+  - If the virtual machine is powered off or VMware Tools are not available, then this flag is ignored.
+  - If the virtual machine does not provide capability to take quiesce snapshot, then this flag ignored.
```
Comment (anna-savina) [inline roles/snapshot_management/README.md:110]:
```suggestion
  - If the virtual machine does not provide capability to take quiesce snapshot, then this flag is ignored.
```

File: roles/system_settings/README.md [hunk 6]
```diff
@@ -95,9 +95,6 @@ Note: NTP service will be restarted if configuration is changed.
 - **system_settings_firewall_rules_append**:
   - If `false` the rules overwrites the existing firewall rules and creates a new rule list. If `true` we append the rules to existing rules. Default is `true`.
```
Comment (anna-savina) [inline roles/system_settings/README.md:96]:
```suggestion
  - If `false` the rules overwrite the existing firewall rules and creates a new rule list. If `true` we append the rules to existing rules. Default is `true`.
```

File: roles/provision_vm/README.md [hunk 7]
```diff
@@ -37,729 +39,697 @@ N/A
...
-    Name of the virtual machine to work with.
-    Virtual machine names in vCenter are not necessarily unique, which may be problematic, see O(name_match).
-    If multiple virtual machines with same name exists, then O(folder) is required parameter to
-    identify uniqueness of the virtual machine.
-    This parameter is required, if (state=poweredon), (state=powered-on), (state=poweredoff), (state=powered-off),
-    (state=present), (state=restarted), (state=suspended) and virtual machine does not exists.
-    This parameter is case sensitive.
+    - Name of the virtual machine to manage.
+    - Virtual machine names in vCenter are not necessarily unique which may be problematic.
+      If multiple virtual machines with same name exists, then `provision_vm_folder` is required parameter to
```
Comment (anna-savina) [inline roles/provision_vm/README.md:44]:
```suggestion
      If multiple virtual machines with the same name exist, then `provision_vm_folder` is required parameter to
```

File: roles/provision_vm/README.md [hunk 8]
```diff
@@ -37,729 +39,697 @@ N/A
...
 
 - **provision_vm_uuid**  (string):
-    UUID of the virtual machine to manage if known, this is VMware's unique identifier.
-    This is required if O(name) is not supplied.
-    If virtual machine does not exists, then this parameter is ignored.
-    Please note that a supplied UUID will be ignored on virtual machine creation, as VMware creates the UUID internally.
+    - The instance UUID of the virtual machine to manage.
+    - This is required if `provision_vm_name` is not supplied.
+    - Note that a supplied UUID will be ignored on virtual machine creation, as VMware creates the UUID internally.
+      If virtual machine does not exists, then this parameter is ignored.
```
Comment (anna-savina) [inline roles/provision_vm/README.md:53]:
```suggestion
      If virtual machine does not exist, then this parameter is ignored.
```

File: roles/provision_vm/README.md [hunk 9]
```diff
@@ -37,729 +39,697 @@ N/A
...
-        - Type of disk mode.
-        - If V(persistent) specified, changes are immediately and permanently written to the virtual disk. This is default.
-        - If V(independent_persistent) specified, same as persistent, but not affected by snapshots.
-        - If V(independent_nonpersistent) specified, changes to virtual disk are made to a redo log and discarded at power off,
-          but not affected by snapshots.
+        - type: str
+        - choices: ['persistent', 'independent_persistent', 'independent_nonpersistent']
+        - description:
+            - Type of disk mode.
+            - If `persistent` specified, changes are immediately and permanently written to the virtual disk. This is default.
```
Comment (anna-savina) [inline roles/provision_vm/README.md:345]:
```suggestion
            - If `persistent` is specified, changes are immediately and permanently written to the virtual disk. This is default.
```

File: roles/provision_vm/README.md [hunk 10]
```diff
@@ -37,729 +39,697 @@ N/A
...
+            - If `independent_persistent` specified, same as persistent, but not affected by snapshots.
```
Comment (anna-savina) [inline roles/provision_vm/README.md:346]:
```suggestion
            - If `independent_persistent` is specified, same as persistent, but not affected by snapshots.
```

File: roles/provision_vm/README.md [hunk 11]
```diff
@@ -37,729 +39,697 @@ N/A
...
+            - If `independent_nonpersistent` specified, changes to virtual disk are made to a redo log and discarded at power off, but not affected by snapshots.
```
Comment (anna-savina) [inline roles/provision_vm/README.md:347]:
```suggestion
            - If `independent_nonpersistent` is specified, changes to virtual disk are made to a redo log and discarded at power off, but not affected by snapshots.
```

File: roles/provision_vm/README.md [hunk 12]
```diff
@@ -37,729 +39,697 @@ N/A
...
+    - Attributes `controller_type`, `controller_number`, and `unit_number` are used to configure multiple types of disk controllers and disks for creating or reconfiguring virtual machine.
+    - Each item in the list may have the following attributes:
     * size:
-        description:
-        - Disk storage size.
-        - Please specify storage unit like [kb, mb, gb, tb].
-        type: str
+        - description:
+            - Disk storage size.
+            - You should include the storage unit like 100kb, 100mb, etc
```
Comment (anna-savina) [inline roles/provision_vm/README.md:304]:
```suggestion
            - You should include the storage unit such as 100kb, 100mb, etc
```

**Comment by mikemorency** [comment]
@anna-savina 
Other similar collections, like aws_ops, azure_ops, or gcp_ops, do not list python or ansible dependencies in their role docs. They only list those in the collection level docs/galaxy file. Instead, they list system requirements like OS packages or pre-req steps that must be done.

Most of our roles have no requirements like that. I did list some for the vcenter and esxi provisioning roles.

Heres some examples for reference:
https://console.redhat.com/ansible/automation-hub/repo/validated/cloud/gcp_ops/content/role/clone_on_prem_vm/
https://console.redhat.com/ansible/automation-hub/repo/validated/cloud/azure_ops/content/role/azure_load_balancer_with_public_ip/
https://console.redhat.com/ansible/automation-hub/repo/validated/cloud/aws_ops/content/role/awsconfig_apigateway_with_lambda_integration/

File: roles/deploy_ovf/README.md [hunk 13]
```diff
@@ -2,31 +2,31 @@
...
 - **deploy_ovf_hostname**:
-  - str, Required. The hostname or IP address of the vSphere vCenter or ESXI host.
+  - str, Required. The hostname or IP address of the vSphere vCenter or ESXi host.
 
 - **deploy_ovf_validate_certs**
   - bool, Allows connection when SSL certificates are not valid. Set to false when certificates are not trusted.
 
 - **deploy_ovf_port**:
-  - str or int, The port to use to authenticate to the vSphere vCenter or ESXI host.
+  - str or int, The port used to authenticate to the vSphere vCenter that contains the cluster to configure.
```
Comment (anna-savina) [inline roles/deploy_ovf/README.md:24]:
```suggestion
  - str or int, The port used to authenticate to the vSphere vCenter or ESXi host.
```

File: roles/export_vm_as_ovf/README.md [hunk 14]
```diff
@@ -27,14 +27,14 @@ N/A
   - The name of the datacenter in vSphere vCenter which contains the VM.
 
 - **export_vm_as_ovf_port**:
-  - str or int, The port to use to authenticate to the vSphere vCenter which contains the VM.
+  - str or int, The port used to authenticate to the vSphere vCenter that contains the cluster to configure.
```
Comment (anna-savina) [inline roles/export_vm_as_ovf/README.md:30]:
```suggestion
  - str or int, The port used to authenticate to the vSphere vCenter that contains the VM.
```

File: roles/provision_vm/README.md [hunk 15]
```diff
@@ -37,729 +39,697 @@ N/A
...
-        description:
-        - Type of disk controller.
-          Set this type on not supported ESXi or VM hardware version will lead to failure in deployment.
-        - When set to V(sata), please make sure O(disk.unit_number) is correct and not used by SATA CDROMs.
-        - If set to V(sata) type, please make sure O(disk.controller_number) and O(disk.unit_number) are set correctly when O(cdrom=sata).
+        - type: str
+        - choices: ['buslogic', 'lsilogic', 'lsilogicsas', 'paravirtual', 'sata', 'nvme']
+        - description:
+            - Type of disk controller.
+            - Set this type on not supported ESXi or VM hardware version will lead to failure in deployment.
```
Comment (anna-savina) [inline roles/provision_vm/README.md:353]:
```suggestion
            - Setting this type on not supported ESXi or VM hardware version will lead to deployment failure.
```

File: roles/provision_vm/README.md [hunk 16]
```diff
@@ -37,729 +39,697 @@ N/A
...
 
 - **provision_vm_force** (boolean):
-    Ignore warnings and complete the actions.
-    This parameter is useful while removing virtual machine which is powered on state.
-    This module reflects the VMware vCenter API and UI workflow, as such, in some cases the `force` flag will be mandatory to perform the action to ensure you are certain the action has to be taken, no matter what the consequence. This is specifically the case for removing a powered on the virtual machine when state=absent.
-    Choices:
-    - false ← (default)
-    - true
+    - Ignore warnings and complete the actions.
+    - This parameter is useful while removing virtual machine which is powered on state.
```
Comment (anna-savina) [inline roles/provision_vm/README.md:386]:
```suggestion
    - This parameter is useful while removing virtual machine that is in powered on state.
```

File: roles/provision_vm/README.md [hunk 17]
```diff
@@ -37,729 +39,697 @@ N/A
...
-        - The firmware of virtual machine must be EFI and secure boot must be enabled.
-        - Virtualization Based Security depends on nested virtualization and Intel Virtualization Technology for Directed I/O.
-        - Deploy on unsupported ESXi, hardware version or firmware may lead to failure or deployed VM with unexpected configurations.
+        - type: bool
+        - description:
+            - Enable Virtualization Based Security feature for Windows on ESXi 6.7 and later, from hardware version 14.
+            - Supported Guest OS are Windows 10 64 bit, Windows Server 2016, Windows Server 2019 and later.
+            - The firmware of virtual machine must be EFI and secure boot must be enabled.
+            - Virtualization Based Security depends on nested virtualization and Intel Virtualization Technology for Directed I/O.
+            - Deploy on unsupported ESXi, hardware version or firmware may lead to failure or deployed VM with unexpected configurations.
```
Comment (anna-savina) [inline roles/provision_vm/README.md:512]:
```suggestion
            - Deploying on unsupported ESXi, hardware version or firmware may lead to failure or deployed VM with unexpected configurations.
```

File: roles/provision_vm/README.md [hunk 18]
```diff
@@ -37,729 +39,697 @@ N/A
...
-         type: str
-         description:
-         - The label of the virtual NVDIMM device to be removed or configured, e.g., "NVDIMM 1".
-         - 'This parameter is required when O(nvdimm.state=absent), or O(nvdimm.state=present) to reconfigure NVDIMM device
-           size. When add a new device, please do not set.'
+        - type: str
+        - description:
+            - The label of the virtual NVDIMM device to be removed or configured, e.g., "NVDIMM 1".
+            - This parameter is required when `provision_vm_nvdimm.state` is `absent` or `present` to reconfigure NVDIMM device
+              size. When add a new device, please do not set.
```
Comment (anna-savina) [inline roles/provision_vm/README.md:715]:
```suggestion
              size. When adding a new device, please do not set it.
```

**Review by anna-savina — APPROVED** [review]
LGTM

**Pull request merged**